test/e2e/search_test.go

```diff
@@ -344,6 +344,55 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
 	ginkgo.Describe("backend store testing", ginkgo.Ordered, func() {
 	})
 
+	ginkgo.Describe("update cluster labels testing", ginkgo.Ordered, func() {
+
+		var (
+			pollTimeout, pollInterval = time.Second * 10, time.Second
+			proxyClient               kubernetes.Interface
+			m1Dynamic                 dynamic.Interface
+			nodeGVR                   = corev1.SchemeGroupVersion.WithResource("nodes")
+		)
+
+		ginkgo.BeforeAll(func() {
+			m1Dynamic = framework.GetClusterDynamicClient(member1)
+			gomega.Expect(m1Dynamic).ShouldNot(gomega.BeNil())
+			proxyConfig := *restConfig
+			proxyConfig.Host += "/apis/search.karmada.io/v1alpha1/proxying/karmada/proxy"
```

Comment:
Hi @tedli, this e2e might not be written correctly. The test coverage does not include the modified content. In this revision, we have added handling for cluster modification events, but this only applies to search cache and not search proxy capability.

Have you tested the new processing logic locally? If it passes the local test, please share the test results. 
And then we can submit an E2E test for the modification in a subsequent PR to promote the integration of the current refactoring code.

Reply:
I'm saying that because this PR has been blocked for weeks.

Reply:
Yes, I think we can start by removing the E2E part and adding a separate PR to include e2e testing. This test can be completed by anyone who is interested. I create Issue #3731 to track it. How do you think @tedli

Reply:
Hi all,
Sorry for late reply. I will remove e2e test case, and I'm glad to take a deeper dive with karmada e2e test, and then finish #3731

Reply:
Thanks @tedli